### ISSUE TITLE
alarm_panel: alarm_code reset on adapter restart?!

### ISSUE THREAD

**Issue by marcprenger** [ioBroker/ioBroker.lovelace#291]
I'm trying to set a custom code (PIN) for the alarm_control_panel card. 

Therefore I changed the "alarm_code" under "native" of object "lovelace.X.control.alarm" (X=0 in my case) and restarted the lovelace.0 adapter ...

My custom alarm_code is being overwritten at every adapter restart.

I'm using latest ioBroker / ioBroker.lovelace ...

**Comment by Garfonso** [comment]
Yes, this sadly is a bug in js-controller, currently. It is already fixed, but seems that it is not yet released, see here: https://github.com/ioBroker/ioBroker.js-controller/issues/1481

Until this is fixed in controller, maybe the best work around is to create your own alarm state somewhere (for exampe 0_userdata) and use the custom configuration and configure it as alarm_control_panel entity manually. It should work just as the `lovelace.X.control.alarm` then (of course the entity Id needs to be unique and different from the build in entity).

**Comment by marcprenger** [comment]
I wrote a small script in ioBroker which sets the alarm-code right after instance start / restart (see below).

As for me, this only works if I define the alarm entity manually:
![image](https://user-images.githubusercontent.com/27813142/148673852-edbd9e26-8402-48c7-9ea4-ffc2b4ab4c9e.png)

Indeed, this throws a warning in log:
![image](https://user-images.githubusercontent.com/27813142/148673905-6bf3c57e-8517-4628-bd55-0403d5695c22.png)
 
`var lovelace_code = 4321; // alarm-code (PIN)
var lovelace_alive_wait = 5000; // wait xxx miliseconds after adapter restart

var id_lovelace_alive = "system.adapter.lovelace.0.alive"; // adapter alive state 
var id_lovelace_alarm = "lovelace.0.control.alarm"; // alarm-panel entity (with alarm-code)

function lovelace_set_code() {
    var lovelace_alarm = getObject(id_lovelace_alarm);

    if (lovelace_alarm.native.alarm_code != lovelace_code) {
        lovelace_alarm.native.alarm_code = lovelace_code;
        setObject(id_lovelace_alarm, lovelace_alarm);

        console.log("setting Lovelace alarm_code=****");
    }
}

on({id: id_lovelace_alive, change: 'ne', val: false}, function () {setTimeout(lovelace_set_code, lovelace_alive_wait)}); // wait for adapter restart
`

**Comment by Garfonso** [comment]
The warning should be solved with 2.1.x (should be available in latest soonish).

I can not really see why the code change should only work with a manual entity. Hm.
